### PR TITLE
Adding annotations on public APIs

### PR DESCRIPTION
**Pull request by emdobrin**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity/IdentityPublicAPITest.java [hunk 1]
```diff
@@ -64,7 +64,23 @@ public void testGetExtensionVersionAPI() {
 	@Test
 	public void testRegisterExtensionAPI() throws InterruptedException {
 		// test
-		// Identity.registerExtension() is called in the setup method
+		//noinspection deprecation
+		Identity.registerExtension();
```
Comment (prudrabhat) [inline code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity/IdentityPublicAPITest.java:68]: My bad. Thanks for catching this!

**Review by prudrabhat — APPROVED** [review]

File: code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/Identity.java [hunk 2]
```diff
@@ -47,29 +46,27 @@ private Identity() {}
...
 	}
 
 	/**
 	 * Registers the extension with the Mobile SDK. This method should be called only once in your application class.
 	 *
-	 * @deprecated Use {@link MobileCore#registerExtensions(List, AdobeCallback)} with {@link Identity#EXTENSION} instead.
+	 * @deprecated as of 2.0.0, use {@link MobileCore#registerExtensions(List, AdobeCallback)} with {@link Identity#EXTENSION} instead.
 	 */
 	@Deprecated
+	@SuppressWarnings("deprecation")
```
Comment (addb) [inline code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/Identity.java:60]: Why are we suppressing warnings here?

Comment (emdobrin) [reply code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/Identity.java:60]: it shows a warning in the IDE when using the registerExtension API so it suppresses that, but you would still see the deprecation message on the API itself

File: code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity/IdentityPublicAPITest.java (hunk 1 above)
Comment (emdobrin) [reply code/edgeidentity/src/androidTest/java/com/adobe/marketing/mobile/edge/identity/IdentityPublicAPITest.java:68]: ahh.. good catch on the catch :)

**Review by prudrabhat — APPROVED** [review]

**Pull request merged**